# Data Analysis and Preprocessing

In [3]:
# import of libraries

import pandas as pd
import numpy as np
import seaborn as sns 
import matplotlib.pyplot as plt 
%matplotlib inline
sns.set(color_codes=True)
import warnings
warnings.filterwarnings('ignore')


In [4]:
# uploading dataset

df = pd.read_csv("student-lpor.csv")

In [5]:
# Dataset visualization

df.head(8)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   at_home   teacher   
1     GP   F   17       U     GT3       T     1     1   at_home     other   
2     GP   F   15       U     LE3       T     1     1   at_home     other   
3     GP   F   15       U     GT3       T     4     2    health  services   
4     GP   F   16       U     GT3       T     3     3     other     other   
5     GP   M   16       U     LE3       T     4     3  services     other   
6     GP   M   16       U     LE3       T     2     2     other     other   
7     GP   F   17       U     GT3       A     4     4     other   teacher   

   ... romantic famrel  freetime  goout Dalc Walc health absences  G1  G2  
0  ...       no      4         3      4    1    1      3        4   0  11  
1  ...       no      5         3      3    1    1      3        2   9  11  
2  ...       no      4         3      2    2    3      3        6  12  13  
3  ...      yes      3         2      2    1    1      5        0  14  14  
4  ...       no      4         3      2    1    2      5        0  11  13  
5  ...       no      5         4      2    1    2      5        6  12  12  
6  ...       no      4         4      4    1    1      3        0  13  12  
7  ...       no      4         1      4    1    1      1        2  10  13  

[8 rows x 31 columns]

In [6]:
# dataset caracteristics 

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 31 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      649 non-null    object
 1   sex         649 non-null    object
 2   age         649 non-null    int64 
 3   address     649 non-null    object
 4   famsize     649 non-null    object
 5   Pstatus     649 non-null    object
 6   Medu        649 non-null    int64 
 7   Fedu        649 non-null    int64 
 8   Mjob        649 non-null    object
 9   Fjob        649 non-null    object
 10  reason      649 non-null    object
 11  guardian    649 non-null    object
 12  traveltime  649 non-null    int64 
 13  studytime   649 non-null    int64 
 14  schoolsup   649 non-null    object
 15  famsup      649 non-null    object
 16  paid        649 non-null    object
 17  activities  649 non-null    object
 18  nursery     649 non-null    object
 19  higher      649 non-null    object
 20  internet  

In [7]:
# dataset size
df.shape

(649, 31)

The dataset has 649 observations and 30 features, with numbers, decimals and objects included

In [8]:
# Check for missing values

df.isnull().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
dtype: int64

there are no missing values

In [9]:
# Checking for duplicates

duplicate_rows_df = df[df.duplicated()]
duplicate_rows_df.shape

(0, 31)

There are no duplicates

In [11]:
# replacing categorical values for numerical values

df['school'].replace({'GP': 0, 'MS': 1}, inplace=True)
df['sex'].replace({'F': 0, 'M': 1}, inplace=True)
df['address'].replace({'U': 0, 'R': 1}, inplace=True)
df['famsize'].replace({'LE3': 0, 'GT3': 1}, inplace=True)
df['Pstatus'].replace({'T': 0, 'A': 1}, inplace=True)
df['Mjob'].replace({'teacher': 0, 'health': 1, 'services': 2, 'at_home': 3, 'other': 4}, inplace=True)
df['Fjob'].replace({'teacher': 0, 'health': 1, 'services': 2, 'at_home': 3, 'other': 4}, inplace=True)
df['reason'].replace({'home': 0, 'reputation': 1, 'course': 2, 'other': 3}, inplace=True)
df['guardian'].replace({'mother': 0, 'father': 1, 'other': 3}, inplace=True)
df['schoolsup'].replace({'yes': 0, 'no': 1}, inplace=True)
df['famsup'].replace({'yes': 0, 'no': 1}, inplace=True)
df['paid'].replace({'yes': 0, 'no': 1}, inplace=True)
df['activities'].replace({'yes': 0, 'no': 1}, inplace=True)
df['nursery'].replace({'yes': 0, 'no': 1}, inplace=True)
df['higher'].replace({'yes': 0, 'no': 1}, inplace=True)
df['internet'].replace({'yes': 0, 'no': 1}, inplace=True)
df['romantic'].replace({'yes': 0, 'no': 1}, inplace=True)

In [12]:
# Checking that the replacement was done correctly

df.head(8)

school  sex  age  address  famsize  Pstatus  Medu  Fedu  Mjob  Fjob  ...  \
0       0    0   18        0        1        1     4     4     3     0  ...   
1       0    0   17        0        1        0     1     1     3     4  ...   
2       0    0   15        0        0        0     1     1     3     4  ...   
3       0    0   15        0        1        0     4     2     1     2  ...   
4       0    0   16        0        1        0     3     3     4     4  ...   
5       0    1   16        0        0        0     4     3     2     4  ...   
6       0    1   16        0        0        0     2     2     4     4  ...   
7       0    0   17        0        1        1     4     4     4     0  ...   

   romantic  famrel  freetime  goout  Dalc  Walc  health  absences  G1  G2  
0         1       4         3      4     1     1       3         4   0  11  
1         1       5         3      3     1     1       3         2   9  11  
2         1       4         3      2     2     3       3         6  12  13  
3         0       3         2      2     1     1       5         0  14  14  
4         1       4         3      2     1     2       5         0  11  13  
5         1       5         4      2     1     2       5         6  12  12  
6         1       4         4      4     1     1       3         0  13  12  
7         1       4         1      4     1     1       1         2  10  13  

[8 rows x 31 columns]

In [13]:
# checking that there are only numerical values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 649 entries, 0 to 648
Data columns (total 31 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   school      649 non-null    int64
 1   sex         649 non-null    int64
 2   age         649 non-null    int64
 3   address     649 non-null    int64
 4   famsize     649 non-null    int64
 5   Pstatus     649 non-null    int64
 6   Medu        649 non-null    int64
 7   Fedu        649 non-null    int64
 8   Mjob        649 non-null    int64
 9   Fjob        649 non-null    int64
 10  reason      649 non-null    int64
 11  guardian    649 non-null    int64
 12  traveltime  649 non-null    int64
 13  studytime   649 non-null    int64
 14  schoolsup   649 non-null    int64
 15  famsup      649 non-null    int64
 16  paid        649 non-null    int64
 17  activities  649 non-null    int64
 18  nursery     649 non-null    int64
 19  higher      649 non-null    int64
 20  internet    649 non-null    int6

All the variables are now numericals